### ISSUE TITLE
After switching to Vue 3 module.data only returns proxy

### ISSUE THREAD

**Issue by trevorgeise** [CyCraft/magnetar#109]
Not sure what I've done here, but my code that was working properly in Vue 2 is now only returning a Proxy(object) with magnetar in Vue 3 and vite.

Here is what it looks like when I console.log myModule.data now
![image](https://user-images.githubusercontent.com/4587981/230720771-1896b3e7-2045-4cab-8a51-a3912edc8133.png)

Here is what it looked like in Vue 2
![image](https://user-images.githubusercontent.com/4587981/230720855-20182339-2bd3-44ce-9a8b-d27b2bd38f28.png)

In Vue 2 I could then call myModule.data.values() and get the values, but in Vue 3 I get
![image](https://user-images.githubusercontent.com/4587981/230720889-bd572b5c-b535-42c8-8c3b-7d3a99eb6e7a.png)

Any ideas?

**Comment by mesqueeb** [comment]
A proxy is an invisible wrapper around the objects / arrays / maps that injects Vue's reactivity. It's because I use Vue 3 `reactive`.

Just like any ref/reactive you console log you'll see the same. 

The thing is, at runtime a proxy is invisible. This is just the way Vue 3 works, not Magnetar related. 

Btw, if you wanna see the inside of what the proxy is wrapping you can click on that 0 to see entry 0 for example.

I suggest you do a quick course on how JS proxies work if you're interested in how Vue 3 works under the hood.

**Comment by mesqueeb** [comment]
Since there is no bug and this is only a console log thing (proxies don't show like that at runtime) 
I'm gonna close this ticket. But feel free to continue the conversation 🙂

**Comment by trevorgeise** [comment]
Thanks for the quick response. I'm definitely knew to proxies, but did dig around a bit. My understanding is I should be able to treat them like a normal object. I'm stuck at how to get to underlying data. In Vue 2 I would do myModule.data.values() and would see the various documents I fetched.  But in Vue 3 I do that and it's this thing:
![image](https://user-images.githubusercontent.com/4587981/230748189-e857788e-f36f-42c7-910e-9acebdac28de.png)

You are right, if I log just myModule.data I can click on the 0 and see my data, I just don't know how to get at it.

**Comment by trevorgeise** [comment]
hmmm.  If I do {... [...temp1.data ][ 0 ][ 1 ] } I can get to the data of the first document in the collection. It's like it's a proxy wrapped in a proxy wrapped in a proxy. I must be doing something weird.

**Comment by mesqueeb** [comment]
The proxy wrapper is not a part of the access path. You just see it in the console. But the actual code and usage of magnetar is **identical** no matter Vue 2 or Vue 3. 

This means, a map you can just do 
```ts
const array = [...map.values()]
```
Even though that map is wrapped in a proxy.

**Comment by trevorgeise** [comment]
What I am finding is the usage is not identical now in Vue 3. Likely I am doing something screwy, but here is what I am finding. 

What I used to do, and what I see in your documentation, is
```const array = module.data.values()```

I should be able to do treat array like a normal array like I used to. I could forEach over it or just call up array[0]. But if I do that now the result is not an array of my data. array[0] equals undefined. forEach is not a function.

However, if I do ```const array = [ ...module.data.values()]``` it is an array of my value. Now array[0] returns a proxy for my actual document, which is great. I can work with that, but it doesn't seem right. I would have rewrite all that code, and I'm pretty sure that would break the reactivity.


Now if I try this with a single document like so
```const doc = module.doc('1234').data ```
It works as expected. I get a proxy back, but I can then treat it like normal. doc.id gives me the id, not undefined.

It's just when I try to access the array of results it gets screwy.
